### PR TITLE
feat: implement a revised version of sda-commons-server-starter

### PR DESCRIPTION
**Pull request by huebner-j**
This change introduces a new version of the `sda-commons-server-starter` module. The change includes the support of OpenApi 3. By integrating this module, only OpenApi 3 annotations are supported in REST endpoint definitions (follow the the links for further details [1][2])
    
Due to the support of OpenApi 3 there are also changes to the `SdaPlatformBundle`. The creation of a dropwizard application differs slightly when using OpenApi 3. The following example shows how an application is created (see [3]):

```
...

@OpenAPIDefinition(
    info =
        @Info(
            title = "SDA Platform Example Application",
            description =
                "This application is an example to show a developer how to create a SDA Dropwizard application",
            version = "1.1.1",
            contact =
                @Contact(name = "John Doe", email = "info@example.com", url = "j.doe@example.com")))
public class SdaPlatformExampleApplication extends Application<SdaPlatformConfiguration> {

  public static void main(String[] args) throws Exception {
    new SdaPlatformExampleApplication().run(args);
  }

  @Override
  public void initialize(Bootstrap<SdaPlatformConfiguration> bootstrap) {
    bootstrap.addBundle(
        SdaPlatformBundle.builder()
            .usingSdaPlatformConfiguration()
            .withRequiredConsumerToken()
            // Additional Swagger documentation properties may be set here until
            // the packages that should be scanned for Swagger documentation annotations are defined
            .addOpenApiResourcePackageClass(this.getClass())
            // or use an existing existing OpenAPI definition.
            // provide the path to the existing openapi file (yaml or json) in the classpath
            .withExistingOpenAPIFromClasspathResource("/custom-openapi.yaml")
            .build());
  }

  ...

```
Hint: Swagger / OpenApi 2 can be used by importing the module `sda-commons-server-starter-swagger`
    
[1] - [https://github.com/SDA-SE/sda-dropwizard-commons/tree/master/sda-commons-server-openapi](https://github.com/SDA-SE/sda-dropwizard-commons/tree/master/sda-commons-server-openapi)
[2] - [https://github.com/swagger-api/swagger-core/wiki/Swagger-2.X---Annotations](https://github.com/swagger-api/swagger-core/wiki/Swagger-2.X---Annotations)
[3] - [https://github.com/swagger-api/swagger-core/wiki/Swagger-2.X---Annotations](https://github.com/SDA-SE/sda-dropwizard-commons/blob/master/sda-commons-server-starter-example/src/main/java/org/sdase/commons/server/starter/example/SdaPlatformExampleApplication.java)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [![Latest Release](https://img.shields.io/github/v/release/sda-se/sda-dropwizard-commons?label=latest)](https://github.com/SDA-SE/sda-dropwizard-commons/releases/latest)
```
Comment (Fox32) [inline README.md:1]: Everything in the readme should reference the new bundle (also the badge)

File: sda-commons-dependency-check/build.gradle [hunk 2]
```diff
@@ -27,7 +27,7 @@ dependencies {
   compile project(':sda-commons-server-s3')
   compile project(':sda-commons-server-s3-testing')
   compile project(':sda-commons-server-security')
-  compile project(':sda-commons-server-starter')
+  compile project(':sda-commons-server-starter-swagger')
```
Comment (Fox32) [inline sda-commons-dependency-check/build.gradle:30]: Please also add the new module to this file. This file is used to detect dependency conflicts

File: sda-commons-server-openapi-example/README.md [hunk 3]
```diff
@@ -5,7 +5,7 @@ This example module shows an
 [`OpenApiBundle`](../sda-commons-server-openapi/src/main/java/org/sdase/commons/server/openapi/OpenApiBundle.java)
 to describe REST endpoints with a OpenApi 3 documentation.
 
-Beside the initialization of the bundle via the [`SdaPlatformBundle`](../sda-commons-server-starter/src/main/java/org/sdase/commons/server/starter/SdaPlatformBundle.java),
+Beside the initialization of the bundle via the [`SdaPlatformBundle`](../sda-commons-server-starter-swagger/src/main/java/org/sdase/commons/server/starter/SdaPlatformBundle.java),
```
Comment (Fox32) [inline sda-commons-server-openapi-example/README.md:8]: I think the new OpenAPI code should reference the new bundle

File: sda-commons-server-openapi-example/build.gradle [hunk 4]
```diff
@@ -1,5 +1,5 @@
 dependencies {
-  compile project(':sda-commons-server-starter')
+  compile project(':sda-commons-server-starter-swagger')
```
Comment (Fox32) [inline sda-commons-server-openapi-example/build.gradle:2]: I think the new OpenAPI code should reference the new bundle

File: sda-commons-server-prometheus-example/build.gradle [hunk 5]
```diff
@@ -1,5 +1,5 @@
 dependencies {
-  compile project(':sda-commons-server-starter')
+  compile project(':sda-commons-server-starter-swagger')
```
Comment (Fox32) [inline sda-commons-server-prometheus-example/build.gradle:2]: Maybe we should use the new bundle everywhere?

File: README.md (hunk 1 above)
Comment (huebner-j) [reply README.md:1]: I think the PR is not ready for review yet. I am actually trying to find all occurrences of `starter-swagger` in parallel..

File: sda-commons-dependency-check/build.gradle (hunk 2 above)
Comment (huebner-j) [reply sda-commons-dependency-check/build.gradle:31]: done

File: sda-commons-server-openapi-example/README.md (hunk 3 above)
Comment (huebner-j) [reply sda-commons-server-openapi-example/README.md]: yes, done

File: sda-commons-server-openapi-example/build.gradle (hunk 4 above)
Comment (huebner-j) [reply sda-commons-server-openapi-example/build.gradle]: done.

File: sda-commons-server-prometheus-example/build.gradle (hunk 5 above)
Comment (huebner-j) [reply sda-commons-server-prometheus-example/build.gradle]: yes, we should use the new bundle everywhere except the `dependency-check` & the `build.gradle` of the root-project. Hopefully I've found every occurrence. Thanks for your comments!

**Comment by huebner-j** [comment]
This PR is closed and reopened under [PR-651](https://github.com/SDA-SE/sda-dropwizard-commons/pull/651) since we decided to implement a new bundle called `sda-commons-starter` and leave the current `sda-commons-server-starter` as is to support backwards compatibility.

**Pull request closed without merging**